subgraphs/abracadabra/src/positions.ts

```diff
@@ -388,3 +202,234 @@ function getAccountBalance(
 
   return tryBalance.reverted ? BIGINT_ZERO : tryBalance.value;
 }
+
+export function getLiquidatePosition(
+  side: string,
+  marketId: string,
+  accountId: string
+): string {
+  const positionCounter = PositionCounter.load(
+    accountId.concat("-").concat(marketId).concat("-").concat(side)
+  );
+  if (!positionCounter) {
+    log.warning("No liquidation position found for account {} on market {}", [
+      accountId,
+      marketId,
+    ]);
+    return "";
+  }
+
+  const position = Position.load(
+    positionCounter.id.concat("-").concat(positionCounter.nextCount.toString())
+  );
+  if (!position) {
+    log.warning("No liquidation position found for account {} on market {}", [
+      accountId,
+      marketId,
+    ]);
+    return "";
+  }
+
+  return position.id;
+}
+
+// A series of side effects on position added
+// They include:
+// * Create a new position when needed or reuse the exisitng position
+// * Update position related data in protocol, market, account
+// * Take position snapshot
+function addPosition(
+  protocol: LendingProtocol,
+  market: Market,
+  account: Account,
+  newBalance: BigInt,
+  eventType: string,
+  side: string,
+  event: ethereum.Event
+): Position {
+  const counterID = account.id
+    .concat("-")
+    .concat(market.id)
+    .concat("-")
+    .concat(side);
+  let positionCounter = PositionCounter.load(counterID);
+  if (!positionCounter) {
+    positionCounter = new PositionCounter(counterID);
+    positionCounter.nextCount = 0;
+    positionCounter.save();
+  }
+  const positionID = positionCounter.id
+    .concat("-")
+    .concat(positionCounter.nextCount.toString());
```

Comment:
Not a big deal, don’t want to delay this PR, but for the next one I’d suggest wrapping all this into its own function. Creating both IDs by concatenating values is something where it is easy to mess up. I think it is worth centralising in one or two functions.